workers/unity/Packages/com.improbable.gdk.tools/.DownloadCoreSdk/Program.cs

```diff
@@ -0,0 +1,133 @@
+                new Package
+                {
+                    Type = "worker_sdk",
+                    Name = "c-dynamic-x86_64-msvc_mt-win32",
+                    TargetPath = $"{nativeDependenciesPath}/Windows/x86_64",
+                    CleanPaths = new List<string> {"include", "worker.lib"}
+                },
+                new Package
+                {
+                    Type = "worker_sdk",
+                    Name = "c-dynamic-x86_64-gcc_libstdcpp-linux",
+                    TargetPath = $"{nativeDependenciesPath}/Linux/x86_64",
+                    CleanPaths = new List<string> {"include"}
+                },
+                new Package
+                {
+                    Type = "worker_sdk",
+                    Name = "c-bundle-x86_64-clang_libcpp-macos",
+                    TargetPath = $"{nativeDependenciesPath}/OSX",
+                    CleanPaths = new List<string> {"include"}
+                },
+                new Package
+                {
+                    Type = "worker_sdk",
+                    Name = "csharp_core",
+                    TargetPath = $"{managedDependenciesPath}"
+                },
+                new Package
+                {
+                    Type = "schema",
+                    Name = "standard_library",
+                    TargetPath = $"{projectPath}/build/dependencies/schema/standard_library"
+                },
+                new Package
+                {
+                    Type = "tools",
+                    Name = "schema_compiler-x86_64-win32",
+                    TargetPath = $"{tempPath}/schema_compiler",
+                    Platform = OSPlatform.Windows
+                },
+                new Package
+                {
+                    Type = "tools",
+                    Name = "schema_compiler-x86_64-macos",
+                    TargetPath = $"{tempPath}/schema_compiler",
+                    Platform = OSPlatform.OSX
+                }
+            };
+
+            try
+            {
+                if (!Directory.Exists(tempPath)) Directory.CreateDirectory(tempPath);
+
+                foreach (var package in Packages.Where(p =>
+                    !p.Platform.HasValue || RuntimeInformation.IsOSPlatform(p.Platform.Value)))
+                {
+                    var tempFilePath = Path.Combine(tempPath, $"{package.Name}.zip");
+                    var newVersion = !File.Exists(tempFilePath);
+                    if (newVersion)
+                    {
+                        Console.Out.WriteLine($"Downloading {package.Name} to {tempFilePath}...");
+
+                        Common.RunRedirected("spatial",
+                            new[] {"package", "retrieve", package.Type, package.Name, coreSdkVersion, tempFilePath});
+                    }
+
+                    if (!newVersion && Directory.Exists(package.TargetPath) &&
+                        Directory.GetFileSystemEntries(package.TargetPath).Any())
+                        continue;
```

Comment:
I'm not sure the skipping the unpack is such a good idea - there are a number of failure cases that can cause weird behaviour.

It can't hurt to re-unpack the files to be safe

Reply:
Every time Unity reloads scripts?...

Reply:
Do we want to codegen every reload?

Reply:
Those are effectively no-ops, so nothing on disk will change and Unity won't re-import.
Unzipping packages unconditionally will ensure that we get stuck in a reload loop since the file times will be different.
Let me see if I can make it easier to deal with...

Reply:
My concern is if that condition true - it doesn't mean that the package is actually there or valid. You can then get into a situation where you have to `git clean -xdf` or manually remove the folder to get back to a correct state.